### PR TITLE
Oxidation api crypto

### LINKED ISSUE #3429: Oxidation finish crypto

### PR DESCRIPTION
**Pull request by TimeEngineer**
# What have changed ?

Migrate all the crypto API

- [ ] Does this PR affect the User ?

closes #3429
depends on #3427

### CODE REVIEW

File: oxidation/libparsec/crates/crypto/src/rustcrypto/secret.rs [hunk 1]
```diff
@@ -13,6 +15,32 @@ use crate::CryptoError;
 
 type Blake2bMac40 = Blake2bMac<U5>;
 
+/// Memory block size in bytes
+const BLOCKSIZE: u32 = 1024;
+/// The maximum amount of RAM that the functions in this module will use, in bytes.
+// https://github.com/sodiumoxide/sodiumoxide/blob/master/libsodium-sys/src/sodium_bindings.rs#L128
+const MEMLIMIT_INTERACTIVE: u32 = 33554432;
+/// The maximum number of computations to perform when using the functions.
+//§ https://github.com/sodiumoxide/sodiumoxide/blob/master/libsodium-sys/src/sodium_bindings.rs#L127
```
Comment (FirelightFlagboy) [inline oxidation/libparsec/crates/crypto/src/rustcrypto/secret.rs:24]:
```suggestion
/// https://github.com/sodiumoxide/sodiumoxide/blob/master/libsodium-sys/src/sodium_bindings.rs#L127
```

File: oxidation/libparsec/crates/crypto/src/rustcrypto/secret.rs [hunk 2]
```diff
@@ -13,6 +15,32 @@ use crate::CryptoError;
 
 type Blake2bMac40 = Blake2bMac<U5>;
 
+/// Memory block size in bytes
+const BLOCKSIZE: u32 = 1024;
+/// The maximum amount of RAM that the functions in this module will use, in bytes.
+// https://github.com/sodiumoxide/sodiumoxide/blob/master/libsodium-sys/src/sodium_bindings.rs#L128
```
Comment (FirelightFlagboy) [inline oxidation/libparsec/crates/crypto/src/rustcrypto/secret.rs:21]:
```suggestion
/// https://github.com/sodiumoxide/sodiumoxide/blob/master/libsodium-sys/src/sodium_bindings.rs#L128
```

File: oxidation/libparsec/crates/crypto/src/rustcrypto/secret.rs [hunk 3]
```diff
@@ -13,6 +15,32 @@ use crate::CryptoError;
...
+const MEMLIMIT_INTERACTIVE: u32 = 33554432;
```
Comment (FirelightFlagboy) [inline oxidation/libparsec/crates/crypto/src/rustcrypto/secret.rs:22]:
```suggestion
const MEMLIMIT_INTERACTIVE: u32 = 33_554_432;
```
Use `_` to make the number more readable

File: parsec/_parsec_pyi/crypto.pyi [hunk 4]
```diff
@@ -9,6 +10,14 @@ class SecretKey:
     def encrypt(self, data: bytes) -> bytes: ...
     def decrypt(self, ciphered: bytes) -> bytes: ...
     def hmac(self, data: bytes, digest_size: int) -> bytes: ...
+    @classmethod
+    def generate_recovery_passphrase(cls) -> Tuple[str, SecretKey]: ...
```
Comment (FirelightFlagboy) [inline parsec/_parsec_pyi/crypto.pyi:14]:
```suggestion
    def generate_recovery_passphrase(cls) -> tuple[str, SecretKey]: ...
```

We don't have to import `Tuple`

**Review by FirelightFlagboy — CHANGES_REQUESTED** [review]

**Review by FirelightFlagboy — APPROVED** [review]

**Review by Max-7 — APPROVED** [review]

**Pull request merged**